deploy/infrastructure/dependencies/terraform-aws-kubernetes/README.md

```diff
@@ -0,0 +1,126 @@
+distributed in worker nodes and a worker node should be at least present in each subnets.
+
+Provisioning is done by annotating a Kubernetes Service resource. The following example deploys a simple http server:
+
+```yaml
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: example
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: example-app
+  namespace: example
+  labels:
+    app: example-app
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: example-app
+  template:
+    metadata:
+      labels:
+        app: example-app
+    spec:
+      containers:
+        - name: nginx
+          image: public.ecr.aws/nginx/nginx:1.21
+          ports:
+            - name: http
+              containerPort: 80
+          imagePullPolicy: IfNotPresent
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: example-service
+  namespace: example
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-type: external
+    service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: ip
+    service.beta.kubernetes.io/aws-load-balancer-scheme: "internet-facing"
+    service.beta.kubernetes.io/aws-load-balancer-ssl-ports: '443'
+    service.beta.kubernetes.io/aws-load-balancer-ssl-cert: [ CERTIFICATE_ARN ]
+    service.beta.kubernetes.io/aws-load-balancer-eip-allocations: [ EIP_ALLOCATION_ID1,EIP_ALLOCATION_ID2,... ]
+    service.beta.kubernetes.io/aws-load-balancer-name: [ LOAD_BALANCER_NAME ]
+    service.beta.kubernetes.io/aws-load-balancer-subnets: [ SUBNET_ID1,SUBNET_ID2,... ]
+spec:
+  selector:
+    app: example-app
+  ports:
+    - port: 443
+      targetPort: 80
+      protocol: TCP
+      name: http
+  type: LoadBalancer
+  loadBalancerClass: service.k8s.aws/nlb
+```
+
+- [CERTIFICATE_ARN]: arn of the certificate managed by AWS Certificate Manager
+- [EIP_ALLOCATION_IDx]: Elastic IP allocation id (The number of elastic IP should equal to the number of SUBNET_IDx)
+- [LOAD_BALANCER_NAME]: Name of the balancer created by the controller
+- [SUBNET_IDx]: Name or ID of a subnet (The number of subnets should equal to the number of EIP_ALLOCATION_IDx)
+
+
+## Test
+
+`terraform apply` generates a resource specification `test-app.yml`. This file can be applied to test a http server
+reachability within the deployed cluster. To apply the resources, follow the next steps:
+
+1. Login to the EKS cluster: `aws eks --region $AWS_REGION update-kubeconfig --name $CLUSTER_NAME`
+2. Create the resources: `kubectl apply -f test-app.yml`
+3. Wait (up to 5min) for the load balancer to be ready. Note that the load balancer may take few minutes to start.
+   Monitor the progress here until the state becomes `Active`: https://console.aws.amazon.com/ec2/home#LoadBalancers:
+4. Connect to the app_hostname and contemplate the nginx default welcome page.
+
+## Clean up
+
+1. Delete all services in kubernetes by running `kubectl delete -f test-app.yml`. 
```

Comment:
I think this step probably belongs in a subsection of the major Test section rather than here (if someone didn't want to run the Test, presumably they wouldn't need to do this to clean up)